#165305222 Users should be able to reset password.

katherine95 opened the pull request:
#### What does this PR do?
- Implements reset password functionality.

#### Description of Task to be completed?
- Develop user interface for the user interface to allow a user to reset the password.
- Redirect URL sent to user's email to reset password confirmation user interface.

#### How should this be manually tested?
- Clone the repository:
    $ git clone https://github.com/andela/ah-the-jedi-frontend.git

- change directory to the ah-the-jedi-frontend directory:
    $ cd ah-the-jedi-frontend/

- Get all branches:
    $ git pull

- Switch to the feature branch:
    $ ft-user-reset-password-165305222

- Install the project dependencies
    $  npm install

- Run the project
    $ npm run start:dev

- Navigate to reset password page
    $<your-localhost>/reset_password

#### What are the relevant pivotal tracker stories?
[#165305222](https://www.pivotaltracker.com/story/show/165305222)
#### Any background context you want to add?
None
#### Screenshots

#### Successful password reset
<img width="1680" alt="Screenshot 2019-06-05 at 14 57 02" src="https://user-images.githubusercontent.com/45317755/58954641-6f1af480-87a2-11e9-9cb9-9dc6ec7106ac.png">


#### Successful password confirm
<img width="1677" alt="Screenshot 2019-06-05 at 14 57 39" src="https://user-images.githubusercontent.com/45317755/58954662-7cd07a00-87a2-11e9-895d-29064ecd6386.png">


#### Passwords don't match
<img width="1664" alt="Screenshot 2019-06-05 at 14 57 49" src="https://user-images.githubusercontent.com/45317755/58954685-8eb21d00-87a2-11e9-897a-a12b1475f782.png">

#### Already reset password
<img width="1680" alt="Screenshot 2019-06-05 at 14 57 59" src="https://user-images.githubusercontent.com/45317755/58954706-9d98cf80-87a2-11e9-8972-943121325ceb.png">

AlvinMugambi committed:
feat(password-reset): implement password reset
ensure a user can reset their password
[Delivers #165305222]

AlvinMugambi committed:
feat(password-reset): implement password reset
ensure a user can reset their password
[Delivers #165305222]

AlvinMugambi committed:
feat(password-reset): implement password reset
ensure a user can reset their password
[Delivers #165305222]

AlvinMugambi committed:
feat(password-reset): implement password reset
ensure a user can reset their password
[Delivers #165305222]

AlvinMugambi committed:
feat(password-reset): implement password reset
ensure a user can reset their password
[Delivers #165305222]

AlvinMugambi committed:
feat(password-reset): implement password reset
ensure a user can reset their password
[Delivers #165305222]

AlvinMugambi committed:
feat(password-reset): implement password reset
ensure a user can reset their password
[Delivers #165305222]

AlvinMugambi committed:
feat(password-reset): implement password reset
ensure a user can reset their password
[Delivers #165305222]

AlvinMugambi committed:
feat(password-reset): implement password reset
ensure a user can reset their password
[Delivers #165305222]

AlvinMugambi committed:
Merge branch 'ft-reset-password-165305222' of https://github.com/andela/ah-the-jedi-frontend into ft-reset-password-165305222

katherine95 committed:
Merge branch 'ft-reset-password-165305222' of https://github.com/andela/ah-the-jedi-frontend into ft-reset-password-165305222

katherine95 committed:
Merge branch 'ft-reset-password-165305222' of https://github.com/andela/ah-the-jedi-frontend into ft-reset-password-165305222

Benkimeric committed:
feat(login): implement user login
- implement user login functionality
- add tests for the login functionalities

[Delivers #165305223]

feat(password-reset): implement password reset
ensure a user can reset their password
[Delivers #165305222]

feat(password-reset): implement password reset
ensure a user can reset their password
[Delivers #165305222]

feat(password-reset): implement password reset
ensure a user can reset their password
[Delivers #165305222]

feat(password-reset): implement password reset
ensure a user can reset their password
[Delivers #165305222]

feat(password-reset): implement password reset
ensure a user can reset their password
[Delivers #165305222]

feat(password-reset): implement password reset
ensure a user can reset their password
[Delivers #165305222]

feat(password-reset): implement password reset
ensure a user can reset their password
[Delivers #165305222]

feat(password-reset): implement password reset
ensure a user can reset their password
[Delivers #165305222]

Benkimeric committed:
feat(login): implement user login
- implement user login functionality
- add tests for the login functionalities

[Delivers #165305223]

ElishaMisoi committed:
feat(signup): Users should be able to signup

-add sign up component
-add activation component
-add tests

[Finishes #165305221]

ElishaMisoi committed:
feat(signup): Users should be able to signup

-add sign up component
-add activation component
-add tests

[Finishes #165305221]

katherine95 committed:
feat(password-reset): implement password reset

- ensure a user can reset their password
[Delivers #165305222]

Commit:
feat(password-reset): implement password reset

- ensure a user can reset their password
[Delivers #165305222]

AlvinMugambi committed:
feat(reset_password):Implement reset password
- ensure a user can reset their password
[Delivers #165305222]

jkamz reviewed (changes requested):
Hi. Please squash your commits. Thanks.

Benkimeric commented on .vscode/launch.json:
```diff
@@ -0,0 +1,14 @@
+{
```
Could you kindly elaborate what is the purpose of this file?

AlvinMugambi replied:
It was automatically updated with the text editor, vscode which was unnecessary. Unnecessary files are now removed.

Benkimeric reviewed (commented):
Nice implementation on resetting a user password. However, would you kindly centre the elements like in this image? I think it will look nicer. 
![image](https://user-images.githubusercontent.com/19903559/58948439-cfa23580-8792-11e9-9057-6410360213c6.png)

AlvinMugambi committed:
feat(login): implement user login
    - implement user login functionality
    - add tests for the login functionalities

    [Delivers #165305223]

AlvinMugambi committed:
Delete launch.json

AlvinMugambi committed:
Delete settings.json

AlvinMugambi commented:
> Nice implementation on resetting a user password. However, would you kindly centre the elements like in this image? I think it will look nicer.
> ![image](https://user-images.githubusercontent.com/19903559/58948439-cfa23580-8792-11e9-9057-6410360213c6.png)

Updated

AnguleMathias committed:
Squash commit
